lang/cs/Org.Apache.REEF.Client/AzureBatch/Storage/AzureStorageClient.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Org.Apache.REEF.Client.AzureBatch.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Client.AzureBatch.Storage
+{
+    internal sealed class AzureStorageClient
+    {
+        private static readonly Logger LOGGER = Logger.GetLogger(typeof(AzureStorageClient));
+        private static readonly string StorageConnectionStringFormat = "DefaultEndpointsProtocol=https;AccountName={0};AccountKey={1}";
+        private static readonly int SASTokenValidityMinutes = 60;
+
+        private readonly string _storageAccountName;
```

Comment:
There are similarities here between the Azure Blob code and the AzureCloudBlockBlob. Could the AzureBlockBlob replace the Azure Blob code?

Reply:
I am not quite following what you mean here. Do you suggest using Org.Apache.REEF.IO.FileSystem.AzureBlob.AzureCloudBlockBlob ? That one seems to be designed for testing purpose. Other than that, this client is also generating a 60-min SAS token for authentication.

Reply:
Ok, thanks for the clarification.